### PR TITLE
[ci][automation/2] complete ci state machine

### PR DESCRIPTION
**Pull request by can-anyscale**
Complete CI state machine with movement from PASSING to FLAKY and FLAKY to JAILED

Test:
- CI

### CODE REVIEW

File: release/ray_release/test_automation/ci_state_machine.py [hunk 1]
```diff
@@ -46,24 +63,39 @@ def _create_github_issue(self) -> None:
             "triage",
             self.test.get_oncall(),
         ]
+        recent_failures = [
+            result for result in self.test_results if result.is_failing()
+        ][:3]
```
Comment (aslonnie) [inline release/ray_release/test_automation/ci_state_machine.py:68]: what does this `[:3]` do? why it is 3?

File: release/ray_release/test_automation/ci_state_machine.py [hunk 2]
```diff
@@ -46,24 +63,39 @@ def _create_github_issue(self) -> None:
...
         return False
 
     def _passing_to_flaky(self) -> bool:
-        # TODO(can): if flaky percentage is too high, to be implemented
+        flake = 0
+        for i in range(1, len(self.test_results) - 1):
+            if (
+                self.test_results[i].is_failing()
+                and self.test_results[i + 1].is_passing()
+                and self.test_results[i - 1].is_passing()
```
Comment (aslonnie) [inline release/ray_release/test_automation/ci_state_machine.py:93]: this means `PPFFPP` will count as no flaky? does not feel quite right to me.

File: release/ray_release/test_automation/ci_state_machine.py (hunk 1 above)
Comment (can-anyscale) [reply release/ray_release/test_automation/ci_state_machine.py:68]: let me put in a constant, the goal is to provide a few instances so test owner can debug them

File: release/ray_release/test_automation/ci_state_machine.py (hunk 2 above)
Comment (can-anyscale) [reply release/ray_release/test_automation/ci_state_machine.py:93]: yes, there can be more patterns, let expand it more as we go; for now, i review go/flaky and think that this pattern is good enough to catch what we want

**Review by aslonnie — APPROVED** [review]
let's see how this goes~ will be easy to change anyways I suppose.

**Pull request merged**